pkg/controller/cluster/main-controller.go

```diff
@@ -805,62 +925,56 @@ func (c *Controller) syncHandler(key string) error {
 			return fmt.Errorf("MinIO doesn't seem to have enough quorum to proceed with binary update")
 		}
 
-		klog.Infof("Attempting Tenant %s MinIO server version %s, to: %s", name, images[0], mi.Spec.Image)
-
 		// Images different with the newer state change, continue to verify
 		// if upgrade is possible
 		mi, err = c.updateTenantStatus(ctx, mi, statusUpdatingMinIOVersion, totalReplicas)
 		if err != nil {
 			return err
 		}
 
-		imageSplits := strings.Split(mi.Spec.Image, ":")
-		if len(imageSplits) == 1 {
-			return fmt.Errorf("MinIO operator does not allow images without RELEASE tags")
-		}
+		klog.V(4).Infof("Collecting artifacts for Tenant '%s' to update MinIO from: %s, to: %s",
+			name, images[0], mi.Spec.Image)
 
-		latest, err := miniov1.ReleaseTagToReleaseTime(imageSplits[1])
+		latest, err := c.fetchArtifacts(mi.Spec.Image)
 		if err != nil {
-			return fmt.Errorf("Unsupported release tag, unable to apply requested update %w", err)
-		}
-
-		currentImageSplits := strings.Split(images[0], ":")
-		if len(currentImageSplits) == 1 {
-			return fmt.Errorf("MinIO operator already deployed container with RELEASE tags, update not allowed please manually fix this using 'kubectl patch --help'")
+			_ = c.removeArtifacts()
+			return err
 		}
 
-		current, err := miniov1.ReleaseTagToReleaseTime(currentImageSplits[1])
+		// FIXME: we can make operator TLS configurable here.
+		updateURL, err := mi.UpdateURL(latest, fmt.Sprintf("http://operator.%s.svc.%s:%s%s",
+			miniov1.GetNSFromFile(), miniov1.ClusterDomain,
+			miniov1.WebhookDefaultPort, miniov1.WebhookAPIUpdate,
+		))
 		if err != nil {
-			return fmt.Errorf("Unsupported release tag on current image, non-disruptive update not allowed %w", err)
-		}
-
-		// Verify if the new release tag is latest, if its not latest refuse to apply the new config.
-		if latest.Before(current) {
-			return fmt.Errorf("Refusing to downgrade the tenant %s to version %s, from %s",
-				name, mi.Spec.Image, images[0])
-		}
+			_ = c.removeArtifacts()
 
-		klog.V(4).Infof("Updating Tenant %s MinIO server version from: %s, to: %s",
-			name, images[0], mi.Spec.Image)
-
-		// FIXME: add provisions to provide a custom update URL for private environments.
-		updateURL, err := mi.UpdateURL(latest, "")
-		if err != nil {
 			// Correct URL could not be obtained, not proceeding to update.
-			return fmt.Errorf("Unable to get canonical update URL, failed with %w", err)
+			return fmt.Errorf("Unable to get canonical update URL for Tenant '%s', failed with %w", name, err)
```

Comment:
I'm seeing through all this error returns, I think maybe we should also update the status of the Tenant to indicate the Update failed for X reason